libs/agno/agno/vectordb/weaviate/weaviate.py

```diff
@@ -130,6 +173,32 @@ def doc_exists(self, document: Document) -> bool:
         collection = self.get_client().collections.get(self.collection)
         return collection.data.exists(doc_uuid)
 
+    async def async_doc_exists(self, document: Document) -> bool:
+        """
+        Validate if the document exists using consistent UUID generation asynchronously.
+
+        Args:
+            document (Document): Document to validate
+
+        Returns:
+            bool: True if the document exists, False otherwise
+        """
+        if not document or not document.content:
+            logger.warning("Invalid document: Missing content.")
+            return False  # Early exit for invalid input
+
+        cleaned_content = document.content.replace("\x00", "\ufffd")
+        content_hash = md5(cleaned_content.encode()).hexdigest()
+        doc_uuid = uuid.UUID(hex=content_hash[:32])
+
+        client = self.get_async_client()
+        await client.connect()
```

Comment:
Would it be possible to only connect once, the first time you get the async client?

Reply:
I guess not, because you don't know when to close again

Reply:
Although we don't seem to do it for the sync one?